### ISSUE TITLE
Null configurator

### ISSUE THREAD

**Issue by jsha** [letsencrypt/letsencrypt#438]
For integration tests between the client and server, I'd like a fully command-line invocation of the client that uses the standalone authenticator and outputs a certificate to a file, without trying to configure a web server. It should also be able to run as non-root by binding to port 5001 instead of 443, and accept an HTTP URL for its Boulder instance.

This is the closest I was able to get, but this still wants to touch some of the web server files, and errors out:
D=`mktemp -d`; chmod 0755 $D  ;  letsencrypt --key-dir $D --work-dir $D --cert-dir $D --config-dir $D --backup-dir $D --cert-path $D/cert.pem --chain-path $D/chain.pem --nginx-mod-ssl-conf $D/nginx.conf -i apache -e -m foo@bar.com -d bar.com -s localhost:4000/acme/new-reg -v -t run

**Comment by kuba** [comment]
After pulling in #400 (`--server` without HTTPS and `--test-mode`) you should be able to just call:
```sh
D=mktemp -d
letsencrypt --config-dir $D --work-dir $D  \
  --text --agree-tos \
  --domain bar.com -m foo@bar.com \
  --test-mode --server localhost:4000/acme/new-reg \
  --authenticator standalone \
  -vvvvvvvv \
  auth
```

The only missing piece is binding to port other than `443`.

**Comment by jsha** [comment]
When I run that I get this error:

```
AttributeError: 'Namespace' object has no attribute 'server_url'
```

**Comment by kuba** [comment]
```sh
D=mktemp -d
mkdir $D/conf $D/work
letsencrypt --config-dir $D/conf --work-dir $D/work \
  --text --agree-tos \
  --domain bar.com -m foo@bar.com \
  --server http://localhost:4000/acme/new-reg \
  --no-verify-ssl --dvsni-port 5001 \
  --authenticator standalone \
  -vvvvvvvv \
  auth
```

is the way to go with the current `master` (modulo #466). However, renewer still hardcodes `/etc/letsencrypt` and `/var/lib/letsencrypt` (#469) :(

**Comment by kuba** [comment]
Should be fixed in #473.
